src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,414 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+
+```python
+#%job_fork_env([language=SCRIPT_LANGUAGE], [path=./FORK_ENV_FILE.py], [force=on/off])
+```
+
+The `force` parameter says if the pragma has to overwrite the task selection scripts or fork environment already set or not.
+
+===== Adding pre and/or post scripts
+
+Sometimes, specified scripts has to be executed before and/or after a particular task. To do that, the solution provides
+ `pre_script` and `post_script` pragmas.
+
+To add a pre-script to a task, please use:
+
+```python
+#%pre_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./PRE_SCRIPT_FILE.py])
+```
+
+To add a post-script to a task, please use:
+
+```python
+#%post_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./POST_SCRIPT_FILE.py])
+```
+
+===== Create a job
+
+To create a job, please use the `#%job()` pragma:
+
+```python
+#%job(name=JOB_NAME)
+```
+
+If the job was already been created, the call of this pragma would just rename the job already created by the new provided name.
+
+**Notice that it is not necessary to create and name explicitly the job. If not done by the user, this step is implicitly
+performed when the job is submitted (check section <<Submit your job to the scheduler>> for more information).**
+
+===== Plot job
+
+To verify the created workflow, please use the `#%draw_job()` pragma to plot it into a separate window:
+
+```python
+#%draw_job()
+```
+
+Two optional parameters can be used to configure the way the kernel plot the workflow.
+
+**inline plotting:**
+
+If this parameter is set to 'off', the workflow plotting is made through the https://matplotlib.org/[Matplotlib]
+external window. The default value is 'on'.
+
+```python
+#%draw_job(inline=off)
+```
+
+**saving into hard disk:**
+
+To be sure the workflow is saved into a _.png_ file, this option needs to be set to *on*. The default value is *off*.
+
+```python
+#%draw_job(save=on)
+```
+
+Note that the job will be named (in the order of existence) by the name provided using the 'name' parameter, by the name of the job
+if it is created, by the name of the notebook if reachable or at worst by "Unnamed_job".
+
+```python
+#%draw_job([name=JOB_NAME], [inline=off], [save=on])
+```
+
+===== Save workflow in dot format
+
+To save the created workflow into a https://www.graphviz.org/[GraphViz] _.dot_ format, please use the `#%write_dot()` pragma:
+
+```python
+#%write_dot(name=FILE_NAME)
+```
+
+
+===== Submit your job to the scheduler
+
+To submit the job to the proactive scheduler, the user has to use the `#%submit_job()` pragma:
+
+```python
+#%submit_job()
+```
+
+If the job is not created, or is not up-to-date, the `#%submit_job()` starts by creating a new job named as the old one.
+To provide a new name, please use the same pragma and provide a name as parameter:
+
+```python
+#%submit_job(name=JOB_NAME)
+```
+
+If the kernel, during its execution, never received a job name, he uses the current notebook name, if possible, or gives a random one.
```

Comment:
I think that this description is already mentioned above.

Reply:
Yes, but it is better to re-explain this point because it could have been different. In the later explanation we have just speak about the `draw_graph`'s behavior.

Reply:
I changed the description as follow:
```We recall that if the job's name is not set, the ProActive kernel uses the current notebook name, if possible, or gives a random one.
```

Reply:
okey